manager/state/raft/raft.go

```diff
@@ -361,7 +361,7 @@ func (n *Node) JoinAndStart(ctx context.Context) (err error) {
 		if err != nil {
 			n.stopMu.Lock()
 			// to shutdown transport
-			close(n.stopped)
+			n.cancelFunc()
```

Comment:
[`n.cancelFunc` already calls `close(n.stopped)`, and ensures that it is only called once, because closing it twice causes a panic]((https://github.com/aaronlehmann/swarmkit/blob/347814599208dd88dba462da56bdad4f1e1f441f/manager/state/raft/raft.go#L238)

This would make it safer to call `manager.Stop()` even after `JoinAndStart` errors, because if `JoinAndStart` errors the channel is closed, and `manager.Stop()` calls `raft.Node.Cancel()`, which calls `n.cancelFunc`.